pyiron_atomistics/atomistics/structure/analyse.py

```diff
@@ -47,6 +76,272 @@ def get_average_of_unique_labels(labels, values):
+        """
+        if self._neigh is None:
+            self._neigh = self.structure.get_neighborhood(
+                self.positions, num_neighbors=self.num_neighbors
+            )
+        return self._neigh
+
+    @property
+    def positions(self):
+        """
+        Positions of the interstitial candidates (and not those of the atoms).
+
+        IMPORTANT: Do not set positions via numpy setter, i.e.
+
+        BAD:
+        ```
+        >>> neigh.positions[0][0] = x
+        ```
+
+        GOOD:
+        ```
+        >>> positions = neigh.positions
+        >>> positions[0][0] = x
+        >>> neigh.positions = positions
+        ```
+
+        This is because in the first case related properties (most importantly the neighborhood
+        information) is not updated, which might lead to inconsistencies.
+        """
+        return self._positions
+
+    @positions.setter
+    def positions(self, x):
+        self.reset()
+        self._positions = x
+
+    @property
+    def hull(self):
+        """
+        Convex hull of each atom. It is mainly used for the volume and area calculation of each
+        interstitial candidate. For more info, see `get_volume` and `get_area`.
+        """
+        if self._hull is None:
+            self._hull = [ConvexHull(v) for v in self.neigh.vecs]
+        return self._hull
+
+    def _create_gridpoints(self, n_gridpoints_per_angstrom=5):
+        cell = self.structure.get_vertical_length()
+        gridpoints = (n_gridpoints_per_angstrom*cell).astype(int)
+        positions = [np.linspace(0, 1, gridpoints[i], endpoint=False) for i in range(3)]
+        positions = np.meshgrid(*positions)
+        return np.einsum(
+            'ji,nj->ni', self.structure.cell, np.stack(positions, axis=-1).reshape(-1, 3)
+        )
+
+    def _remove_too_close(self):
+        neigh = self.structure.get_neighborhood(self.positions, num_neighbors=1)
+        self.positions = self.positions[neigh.distances.flatten()>self.min_distance]
+
+    def _set_interstitials_to_high_symmetry_points(self):
+        self.positions = self.positions+np.mean(self.neigh.vecs, axis=-2)
+        self.positions = self.structure.get_wrapped_coordinates(self.positions)
+
+    def _kick_out_points(self, variance_buffer=0.01):
+        variance = self.get_variance()
+        min_var = variance.min()
+        self.positions = self.positions[variance<min_var+variance_buffer]
+
+    def _cluster_points(self, eps=0.1):
+        extended_positions, indices = self.structure.get_extended_positions(
+            eps, return_indices=True, positions=self.positions
+        )
+        labels = DBSCAN(eps=eps, min_samples=1).fit_predict(extended_positions)
+        coo = coo_matrix((labels, (np.arange(len(labels)), indices)))
+        labels = coo.max(axis=0).toarray().flatten()
+        self.positions = get_mean_positions(
+            self.positions, self.structure.cell, self.structure.pbc, labels
+        )
+
+    def get_positions(
```

Comment:
Looking over the `positions` setter/getter alongside `get_positions`, I think I would be more comfortable if `positions` was a read-only property that did the full jazz found in `get_positions` when called -- i.e. make the current `get_positions` arguments class attributes, make that method private, cache it for faster return, and have `positions` call `_get_positions` under the hood passing in the class attributes (so the cache works?).

Take that with a serious grain of salt though, since I don't currently understand why you would want to give people write access to `positions` -- but you really explicitly *do* do that, so I suspect I'm still missing something pretty fundamental here.